keras_cv/metrics/coco/coco_pymetric.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import six
+import tensorflow as tf
+from keras.callbacks import Callback
+
+from keras_cv import bounding_box
+from keras_cv.metrics.coco.pycoco_utils import compute_pycoco_metrics
+
+
+class COCOPyMetric(Callback):
+    def __init__(self, bounding_box_format):
+        self.model = None
+        self.bounding_box_format = bounding_box_format
+        super().__init__()
+
+    def set_model(self, model):
+        self.model = model
+
+    def on_epoch_end(self, epoch, logs=None):
+        logs = logs or {}
+
+        dataset = self.model._eval_data_handler._dataset
+        gt = {}
+        preds = {}
+
+        for i, batch in enumerate(dataset):
+            gt_i, preds_i = self._eval_batch(batch, i)
+
+            for k, v in six.iteritems(preds_i):
+                if k not in preds:
+                    preds[k] = [v]
+                else:
+                    preds[k].append(v)
+
+            for k, v in six.iteritems(gt_i):
+                if k not in gt:
+                    gt[k] = [v]
+                else:
+                    gt[k].append(v)
+
+        metrics = compute_pycoco_metrics(gt, preds)
+        # Mark these as validation metrics by prepending a val_ prefix
+        metrics = {"val_" + name: val for name, val in metrics.items()}
+
+        logs.update(metrics)
+
+    def _eval_batch(self, batch, index):
+        images, gt_boxes = batch
```

Comment:
This should be structured as 
images, gt_boxes, gt_classes = batch,
instead of gt_boxes [..., 5]
as the latter should be removed before 0.4

Reply:
I agree that this should change, but I think it needs to change at the same time that we restructure this package-wide. Right now, for it to work as a callback it needs to accept data in the same format that the models do, which (at least for RetinaNet, which is what I'm testing with) is `images, boxes`, where boxes include the class.

I think this also may end up as something like
```
images, annotated_boxes = batch
boxes, classes = annotated_boxes
```
if we end up going with the nested concept of box classes (idk where we are on that conversation)

Reply:
+1 to waiting to restructure on a package wide basis.

Reply:
Hmmm, both the RetinaNet and FasterRCNN pipeline does not use any KPLs, so it's actually returning boxes and classes, and our model prediction returns boxes and classes separately, so I'm not sure what's blocking this?

Reply:
I believe we agreed that we should use : 
```
for examples in train_ds:
    image, gt_boxes, gt_classes = examples
```

This is the most native format for any detection workflow.

Reply:
Are we sure we want to diverge?  right now we have all bounding boxes in one format, and MaskRCNN in another format.